sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/storage/blob/AzureStorageBlobAutoConfigurationTests.java

```diff
@@ -230,6 +230,46 @@ void configurationPropertiesShouldBind() {
             });
     }
 
+    @Test
+    void connectionDetailsShouldBind() {
+        String accountName = "test-account-name";
+        String connectionString = String.format(STORAGE_CONNECTION_STRING_PATTERN, accountName, "test-key");
+        String endpoint = String.format("https://%s.blob.core.windows.net", accountName);
+        String customerProvidedKey = "fakekey";
+        this.contextRunner
+            .withPropertyValues(
+                "spring.cloud.azure.storage.blob.endpoint=" + endpoint,
+                "spring.cloud.azure.storage.blob.account-key=test-key",
+                "spring.cloud.azure.storage.blob.sas-token=test-sas-token",
```

Comment:
Add `spring.cloud.azure.storage.blob.connection-string=another-value` to confirm that `AzureStorageBlobConnectionDetails` has higher priority.